### PR TITLE
Add NonMaxSupression operator

### PR DESCRIPTION
**Pull request by HectorSVC**
Add schema for NonMaxSupression operator

### CODE REVIEW

File: onnx/defs/nn/defs.cc [hunk 1]
```diff
@@ -1398,4 +1398,96 @@ ONNX_OPERATOR_SET_SCHEMA(
...
+            0.0f)
+        .Attr(
+            "score_threshold",
+            "Float tensor representing the threshold for deciding when to remove boxes based on score.",
+            AttributeProto::FLOAT)
+        .Attr(
+            "pad_to_max_output_size",
+            "Optional. 1(true) - the output selected_indices is padded to be of length max_output_size. Defaults to 0(false).",
+            AttributeProto::INT,
+            OPTIONAL)
```
Comment (linkerzhang) [inline onnx/defs/nn/defs.cc]: put default value here please.

File: onnx/defs/nn/defs.cc [hunk 2]
```diff
@@ -1398,4 +1398,96 @@ ONNX_OPERATOR_SET_SCHEMA(
...
+            OpSchema::Optional)
+        .TypeConstraint(
+            "T1",
+            {"tensor(float)"},
+            "Constrain input "
+            " types to float tensors.")
+        .TypeConstraint(
+            "T2",
+            {"tensor(int32)"},
+            "Constrain input "
```
Comment (linkerzhang) [inline onnx/defs/nn/defs.cc]: output type.

File: onnx/defs/nn/defs.cc [hunk 3]
```diff
@@ -1398,4 +1398,96 @@ ONNX_OPERATOR_SET_SCHEMA(
...
+            "T1")
+        .Output(
+            0,
+            "selected_indices",
+            "selected indices from the boxes tensor.",
+            "T2")
+        .Output(
+            1,
+            "valid_outputs",
+            "Optional. A 0-D integer tensor representing the number of valid elements in selected_indices, with the valid elements appearing first.",
```
Comment (linkerzhang) [inline onnx/defs/nn/defs.cc]: An integer scalar?

File: onnx/defs/nn/defs.cc [hunk 4]
```diff
@@ -1398,4 +1398,96 @@ ONNX_OPERATOR_SET_SCHEMA(
...
+        .Attr(
+            "pad_to_max_output_size",
+            "Optional. 1(true) - the output selected_indices is padded to be of length max_output_size. Defaults to 0(false).",
+            AttributeProto::INT,
+            OPTIONAL)
+        .Input(
+            0,
+            "boxes",
+            "An input tensor. 2D tensor with shape [num_boxes, 4].",
+            "T1")
```
Comment (linkerzhang) [inline onnx/defs/nn/defs.cc]: you may put "tensor(float)" here directly as T1 defined below only has float.

Comment (linkerzhang) [reply onnx/defs/nn/defs.cc]: Same for the T2 below.

File: onnx/defs/nn/defs.cc (hunk 1 above)
Comment (HectorSVC) [reply onnx/defs/nn/defs.cc]: I didn't find a overload Attr which can accept the (..., bool required, T defaultvalue)

---
In reply to: [242298590](https://github.com/onnx/onnx/pull/1695#discussion_r242298590) [](ancestors = 242298590)

**Comment by linkerzhang** [comment]
@HectorSVC please also add shape inference test.

**Pull request merged**